### PR TITLE
Fix some spelling issues

### PR DESCRIPTION
**Pull request by keeganwitt**
Just some spelling fixes and table reformatting.

### CODE REVIEW

**Review by azdagron — COMMENTED** [review]
Thanks! I saw just one additional thing that needed to be corrected. @MarcosDY , how is this going to play with #3494?

File: CHANGELOG.md [hunk 1]
```diff
@@ -331,7 +331,7 @@
 - SPIRE Server federation configuration in the `federates_with` `bundle_endpoint` block is now deprecated (#2340)
 - SPIRE Server `gcp_iit` NodeAttestor configurable `projectid_whitelist` is deprecated in favor of `projectid_allow_list` (#2253)
 - SPIRE Server `k8s_sat` and `k8s_psat` NodeAttestor configurable `service_account_whitelist` is deprecated in favor of `service_account_allow_list` (#2253)
-- SPIRE Sever `registration_uds_path`/`-registrationUDSPath` configurable and flag has been deprecateed in favor of `socket_path`/`-socketPath` (#2075)
+- SPIRE Sever `registration_uds_path`/`-registrationUDSPath` configurable and flag has been deprecated in favor of `socket_path`/`-socketPath` (#2075)
```
[the remark this answers is not in the document]
Comment (keeganwitt) [reply CHANGELOG.md]: Fixed all of these.

**Comment by keeganwitt** [comment]
> Thanks! I saw just one additional thing that needed to be corrected. @MarcosDY , how is this going to play with #3494?

I know you weren't asking me, but FYI, the formatting I did here makes it match the formatting I previously did in #3474.
I don't see that the linting being worked on in #3494 has any table formatting.

**Review by azdagron — APPROVED** [review]
Thanks much, @keeganwitt !

**Pull request merged**